Consul.Test/SessionTest.cs

```diff
@@ -149,8 +150,8 @@ public async Task Session_Create_RenewPeriodic_TTLExpire()
 
             try
             {
-                var renewTask = _client.Session.RenewPeriodic(TimeSpan.FromSeconds(1), id, WriteOptions.Default, ct);
                 var destroyResponse = await _client.Session.Destroy(id);
```

Comment:
When the current thread didn't manage to call the destroy session subroutine before the session was renewed in the `RenewPeriodic` thread, the test failed. The initial TTL was set to 500 seconds, so the renewal subroutine would wait for another 250 seconds between session renewal. 
To make the test robust, we destroy the session before we even get the chance the other thread to renew the session.

Reply:
Ok. And also, if I understand correctly, one of the key part of your changes that makes sure the server is properly initialized, is that you're creating a session instead of just calling `client.Status.Leader()`, which may not have been enough to ensure the server was fully initialized?

Reply:
Yes, calling the `client.Status.Leader()` wasn't enough, We still got cases where, consul server was reporting to be a leader but the subsequent call to create session was failing with `Unexpected response, status code Forbidden: ACL not found` message.

Reply:
Done